StevesSixLeggedSprinters/BangazonAPI
# Created files for Product Resource

## Status
Choose:
**HOLD**

## Migrations
YES

## Description
I created a product model (Product.cs) in order to create a table in the database for all products. I also created a product controller (ProductController.cs) so we are able to GET a list of products, GET a single product, POST (create) a product, PUT (edit) a specific product, and DELETE a specific product. Also modified BangazonAPIContext file to set up a relationship between the database and the products table. I created a model builder in order to automatically generate a time for when a product is added to the database. In Product.cs, I commented out a line of code that creates a collection of orders because it will tie into the Order Resource that it not yet available to me.


## Todos


## Deploy Notes
Not yet deployable

## Steps to Test or Reproduce
Follow these steps to test out this specific code.

```sh
I have set up the following methods: GET, POST, PUT, DELETE
Follow these step by step in order to test.

git fetch --a  (fetch all branches from GitHub)
git branch -avv (display all branches)
git checkout jk-products (switch into my branch)
git pull origin jk-products (pull my code down to test)
dotnet ef migrations add Initial (migrations need to be built)
dotnet ef database update (update the database since a new migration has been built)
open bangazon.db (open database)
dotnet run

Make sure to have Postman installed: https://www.getpostman.com/
In Postman, 'Post' (add) a couple products to the database to start testing.

Example:
{
        "ProductTypeId": 4,
	"Price": 11.00,
	"Title": "Pizza",
	"Description": "A giant slice of delicious pizza from Joey's.",
	"CustomerId": 1,
	"ProductAmount": 1,
}

To view all products:
Select GET
http://localhost:5000/api/product

To view a single product:
Select GET
http://localhost:5000/api/product/{id}

To edit a single product
Select PUT
http://localhost:5000/api/product/{id}

To delete a single product
Select DELETE
http://localhost:5000/api/product/{id}

```
## Impacted Areas in Application
List general components of the application that this PR will affect:

**Controllers/ProductController.cs**
**Models/Product.cs**
**Data/BangazonAPIContext.cs**

* Reference link to Project Ticket: https://github.com/StevesSixLeggedSprinters/BangazonAPI/issues/2
